### PR TITLE
bears/java: Add InferBear

### PR DESCRIPTION
**Pull request by aachiritoaei**
Adds an InferBear for Java

Fixes https://github.com/coala-analyzer/coala-bears/issues/30

NOTE: My previous PR had some problems with solving conflicts. I've closed that and opened a new one.

### CODE REVIEW

File: bears/java/InferBear.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import re
+
+from coalib.bearlib.abstractions.Lint import Lint
+from coalib.bears.LocalBear import LocalBear
+from coalib.results.RESULT_SEVERITY import RESULT_SEVERITY
+
+
+class InferBear(LocalBear, Lint):
+    executable = 'infer'
+    arguments = '-- javac {filename}'
```
Comment (AbdealiJK) [inline bears/java/InferBear.py]: Is `--` purposely away from `javac` ?

File: bears/java/InferBear.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+        r'(?P<line>.+): '
+        r'(?P<severity>error|warning): '
+        r'(?P<message>.*)')
+    severity_map = {
+        "error": RESULT_SEVERITY.MAJOR,
+        "warning": RESULT_SEVERITY.NORMAL}
+
+    def run(self, filename, file):
+        '''
+        Checks the code with `infer`.
```
Comment (AbdealiJK) [inline bears/java/InferBear.py]: double backticks as this is rst

File: .ci/deps.sh [hunk 3]
```diff
@@ -3,22 +3,24 @@ set -x
 
 # Choose the python versions to install deps for
 case $CIRCLE_NODE_INDEX in
- 0) dep_versions=( "3.3.6" "3.4.3" "3.5.1" ) ;;
+ 0) dep_versions=( "3.3.6" "3.4.3" "3.5.1" "2.7.10" ) ;;
  1) dep_versions=( "3.4.3" ) ;;
  2) dep_versions=( "3.3.6" ) ;;
- *) dep_versions=( "3.5.1" ) ;;
+ *) dep_versions=( "3.5.1" "2.7.10" ) ;;
```
Comment (AbdealiJK) [inline .ci/deps.sh]: What i tried to say in gitter is it should be added at https://github.com/coala-analyzer/coala-bears/pull/229/files#diff-5c64bb36a9fa177d5dfeb57b93e57a6cR49

File: bears/java/InferBear.py (hunk 1 above)
Comment (aachiritoaei) [inline bears/java/InferBear.py]: Yes. That's how infer gets arguments. with a space in between.

File: .ci/deps.sh [hunk 4]
```diff
@@ -67,3 +69,13 @@ julia -e "Pkg.add(\"Lint\")"
...
+
+# Infer commands
+wget -nc -O ~/infer.tar.xz https://github.com/facebook/infer/releases/download/v0.7.0/infer-linux64-v0.7.0.tar.xz
+tar xf ~/infer.tar.xz -C ~/
+cd ~/infer-linux64-v0.7.0
+opam init --y
+opam update
+opam pin add --yes --no-action infer .
+opam install --deps-only --yes infer
+./build-infer.sh java
```
Comment (AbdealiJK) [inline .ci/deps.sh]: Hi, I was checking on how to make this easier.
I found - https://github.com/shaunkawano/circle-ci-android-sample/blob/master/circle.yml#L17
That looks much simpler - can you check it out ?

File: circle.yml [hunk 5]
```diff
@@ -8,11 +8,13 @@ dependencies:
     - ~/.RLibrary
     - ~/dart-sdk/bin
     - ~/.cabal
+    - ~/infer-linux64-v0.7.0
```
Comment (AbdealiJK) [inline circle.yml]: Awesome that you cache it :+1:

File: .ci/deps.sh [hunk 6]
```diff
@@ -44,7 +46,7 @@ bundle install
 
 for dep_version in "${dep_versions[@]}" ; do
   pyenv install -ks $dep_version
-  pyenv local $dep_version
+  pyenv local $dep_version 2.7.10
```
Comment (AbdealiJK) [inline .ci/deps.sh]: awesome :)

File: .ci/deps.sh (hunk 4 above)
Comment (AbdealiJK) [inline .ci/deps.sh]: Just a reminder about that script - I think it'd make things easier ? 
If you think this is fine - let me know

**Comment by Adrianzatreanu** [comment]
I think this is good, all the reviews have been fulfilled and tests passed.

**Comment by Adrianzatreanu** [comment]
ack 6cd3f83

**Comment by Adrianzatreanu** [comment]
@rultor merge

**Pull request merged**